test/e2e/installplan_e2e_test.go

```diff
@@ -3192,14 +3192,24 @@ var _ = Describe("Install Plan", func() {
 
 		It("should clear clear up the condition in the InstallPlan status that contains an error message when a valid OperatorGroup is created", func() {
 
-			// first check that a condition with a message exists
-			fetchedInstallPlan, err := fetchInstallPlanWithNamespace(GinkgoT(), crc, installPlanName, ns.GetName(), buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseInstalling))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(fetchedInstallPlan).NotTo(BeNil())
+			// first wait for a condition with a message exists
 			cond := operatorsv1alpha1.InstallPlanCondition{Type: operatorsv1alpha1.InstallPlanInstalled, Status: corev1.ConditionFalse, Reason: operatorsv1alpha1.InstallPlanReasonInstallCheckFailed,
 				Message: "no operator group found that is managing this namespace"}
-			Expect(fetchedInstallPlan.Status.Phase).To(Equal(operatorsv1alpha1.InstallPlanPhaseInstalling))
-			Expect(hasCondition(fetchedInstallPlan, cond)).To(BeTrue())
+			Eventually(
+				func() (bool, error) {
+					fetchedInstallPlan, err := fetchInstallPlanWithNamespace(GinkgoT(), crc, installPlanName, ns.GetName(), buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseInstalling))
+					Expect(err).NotTo(HaveOccurred())
+					Expect(fetchedInstallPlan).NotTo(BeNil())
+					Expect(fetchedInstallPlan.Status.Phase).To(Equal(operatorsv1alpha1.InstallPlanPhaseInstalling))
+					if hasCondition(fetchedInstallPlan, cond) {
+						return true, nil
+					} else {
+						return false, nil
+					}
+				},
+				1*time.Minute,
+				interval,
+			).Should(BeTrue())
```

Comment:
WDYT about this implementation? I'm not entirely sure whether we care about returning an error here either, and can just return a bool instead?

```suggestion
			Eventually(func() (bool, error) {
				fetchedInstallPlan, err := fetchInstallPlanWithNamespace(GinkgoT(), crc, installPlanName, ns.GetName(), buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseInstalling))
				if err != nil {
					return false, err
				}
				if fetchedInstallPlan.Status.Phase != operatorsv1alpha1.InstallPlanPhaseInstalling {
					return false, nil
				}
				return hasCondition(fetchedInstallPlan, cond), nil
			}, 1*time.Minute, interval).Should(BeTrue())
```

Reply:
It looks better to me. I take your suggested changes and also take out the error return.  Thanks!